internal/extproc/backendauth/api_key.go

```diff
@@ -15,12 +17,17 @@ type apiKeyHandler struct {
 	apiKey string
 }
 
-func newAPIKeyHandler(auth *filterconfig.APIKeyAuth) (Handler, error) {
+func newAPIKeyHandler(auth *filterconfig.APIKeyAuth, logger *slog.Logger) (Handler, error) {
 	secret, err := os.ReadFile(auth.Filename)
 	if err != nil {
 		return nil, fmt.Errorf("failed to read api key file: %w", err)
 	}
-	return &apiKeyHandler{apiKey: string(secret)}, nil
+	trimmedKey := strings.TrimSpace(string(secret))
+	if trimmedKey != string(secret) {
+		logger.Debug("api key contained leading/trailing whitespace and was trimmed", slog.String("filename", auth.Filename))
+	}
```

Comment:
ok, i would rather remove this log

Reply:
Why? I believe it provides valuable debugging information when needed.

Reply:
when you needed?

Reply:
I don’t see a situation where I’d need this information, but since a transformation took place, I think it’s worth mentioning.

That said, it’s a minor change, and I’m fine either way. What’s your final decision?

Reply:
Yep let's remove it for now - we can add it later if anyone complains later!